src/main/java/redis/clients/jedis/ClientSideCache.java

```diff
@@ -0,0 +1,58 @@
+package redis.clients.jedis;
+
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import redis.clients.jedis.exceptions.JedisException;
+import redis.clients.jedis.util.SafeEncoder;
+
+public class ClientSideCache {
+
+  private final Map<ByteBuffer, Object> cache = new HashMap<>();
+
+  protected ClientSideCache() {
+  }
+
+  protected void invalidateKeys(List list) {
```

Comment:
Design discussion:

As I understood `ClientSideCache` object represents a DAO/persistence layer, so it's actually just a storage and you can perform CRUD operations against it. From this perspective this method should be rather named as `delete()` or `batchDelete()`.

Invalidation is a more complex process built around cache that involves business logic and belong to another layer, something like this:
1. Get invalid key (keys).
2. Find all command responses associated with this key (keys).
3. Delete them from cache.

Reply:
We have used Redis terms for easier recognition. Things could be changed later but not now.